core/src/test/kotlin/io/customer/sdk/core/di/EventBusTest.kt

```diff
@@ -27,43 +27,43 @@ class EventBusTest : BaseUnitTest() {
 
     @Test
     fun givenPublishEventVerifySubscribe() = testScope.runTest {
-        val events = mutableListOf<Event.DummyEvent>()
-        val job = eventBus.subscribe<Event.DummyEvent> { event ->
+        val events = mutableListOf<Event>()
+        val job = eventBus.subscribe<Event.ProfileIdentifiedEvent> { event ->
             events.add(event)
         }
 
-        val testEvent = Event.DummyEvent("Test Message")
+        val testEvent = Event.ProfileIdentifiedEvent("Test Message")
         println("Publishing event: $testEvent")
         eventBus.publish(testEvent)
 
         delay(100) // Give some time for the event to be collected
 
         events.size shouldBeEqualTo 1
-        events[0].message shouldBeEqualTo testEvent.message
+        (events[0] as Event.ProfileIdentifiedEvent).identifier shouldBeEqualTo testEvent.identifier
```

Comment:
Instead of casting directly, I think we should have something like this

```suggestion
        events
          .shouldHaveSingleItem()
          .shouldBeInstanceOf<Event.ProfileIdentifiedEvent>()
          .identifier shouldBeEqualTo testEvent.identifier
```